sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -82,52 +82,45 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   object SpecialLimits extends Strategy {
     override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
       case ReturnAnswer(rootPlan) => rootPlan match {
-        case Limit(IntegerLiteral(limit), Sort(order, true, child))
```

Comment:
It seems we missing this case.

Reply:
As I mentioned in the PR description, we don't need to plan `TakeOrderedAndProjectExec` under `ReturnAnswer`, as we don't have special logic for it. It will still be planned in the normal code path, which is `case other => planLater(other) :: Nil` and we do have planner rule to match `Limit(IntegerLiteral(limit), Sort(order, true, child))`

Reply:
handled by `planTakeOrdered`